portal/models/user.py

```diff
@@ -867,6 +867,36 @@ def update_consents(self, consent_list, acting_user):
             db.session.add(replaced)
         db.session.commit()
 
+    def deactivate_tous(self, acting_user, types=None):
+        """ Mark user's current active ToU agreements as inactive
+
+        Marks the user's current active ToU agreements as inactive.
+        User must agree to ToUs again upon next login (per CoreData logic).
+        If types provided, only deactivates agreements of that ToU type.
+        Called when the ToU agreement language is updated.
+
+        :param acting_user: user behind the request for permission checks
+        :param types: ToU types for which to invalide agreements (optional)
+
+        """
+        from .tou import ToU
+
+        for tou in ToU.query.join(Audit).filter(and_(
+                Audit.user_id == self.id,
```

Comment:
Shouldn't this be `Audit.subject_id == self.id`?

Reply:
Ah, good point. Thing is, all the current ToU-setting and ToU-checking logic current checks against `Audit.user_id` (e.g. https://github.com/uwcirg/true_nth_usa_portal/blob/develop/portal/models/coredata.py#L373 ).

Should I update all ToU references to set/check `Audit.subject_id` instead, and also migrate all existing ToUs to set `Audit.subject_id = Audit.user_id`?

Reply:
Lets clean it up!
Aren't those just checking the wrong value.  I mean, it's legit that user_id != subject_id in many flows - but it's only the subject_id that we'd care about when looking for a valid ToU, right?

Reply:
Yea, it looks like we're setting the expected user to `Audit.subject_id` for those ToUs, we just never updated the relevant checks accordingly.

I'll update those (and my own checks here) to look at the correct field.